From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Thu, 15 May 2026 00:00:00 -0400
Subject: [PATCH] Declare tsl concurrency executor dep for custom ops

TF 2.21.0's op_kernel.h (in framework_headers_lib) includes
xla/tsl/concurrency/executor.h. tf_custom_op_library targets did not
declare it, so building custom-op .so files (e.g. the TFLite
microfrontend op) fails Bazel's undeclared-inclusion check. Add
@xla//xla/tsl/concurrency:executor to tf_custom_op_library_additional_deps.
---
diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index 02deea46..a317ac60 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -2203,6 +2203,10 @@ def tf_custom_op_library_additional_deps():
         "@com_google_protobuf//:protobuf_headers",  # copybara:comment
         clean_dep("@eigen_archive//:eigen3"),
         clean_dep("//tensorflow/core:framework_headers_lib"),
+        # framework_headers_lib's op_kernel.h includes xla/tsl/concurrency/
+        # executor.h; declare it so custom-op compiles do not fail Bazel's
+        # undeclared-inclusion check.
+        "@xla//xla/tsl/concurrency:executor",
     ]
 
 # A list of targets that contains the implementation of
